src/components/Sidebar/Sidebar.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  
+<aside 
+
+  class="w-56 absolute inset-y-0 left-0 transform -translate-x-full transition duration-200 ease-in-out  "
+  :class="{'relative -translate-x-0':showSideBar}"
+  aria-label="Sidebar">
+  <div class="overflow-y-auto py-4 px-3  bg-gray-50  h-screen dark:bg-gray-800">
+    <div class="flex items-center justify-between p-2 tracking-widest text-base font-bold text-gray-900 rounded-lg dark:text-white ">
+      <p class="p-[10px]">
+        BeforeClass
+      </p>
+    <button class="hover:bg-gray-100 dark:hover:bg-gray-700 p-2 rounded-lg" @click="showSideBar = !showSideBar">
```

Comment:
extract the `button`s into a standalone component to reduce the code clutter & allow for reusability